cartographer/io/map_io.cc

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2016 The Cartographer Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cartographer/io/map_io.h"
+
+namespace cartographer {
+namespace io {
+
+//constructor takes a const reference to the map_builder
+cartographer::io::MapIO::MapIO(const cartographer::mapping::MapBuilder
+    &map_builder){}
+
+bool MapIO::Write(const string file_name){
```

Comment:
Let's make the serialization part of `MapBuilder`, similar to how there is `SparsePoseGraph::ToProto()`, maybe like this:
```
bool MapBuilder::SerializeState(io::ProtoStreamWriter* writer);
```
There is no need to keep additional state across calls and by putting it at the `MapBuilder` we do not have to expose implementation details.